src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -27,60 +31,83 @@
 
     /**
-     * This method is responsible for the REST GET mechanism to extract the data on
-     * the basis of the subscription name from the Failed Notification Object.
+     * This method is responsible for the REST GET mechanism to extract the data on the basis of the
+     * subscription name from the Failed Notification Object.
      *
      * @param subscriptionName
      */
     @Override
     @ApiOperation(value = "Retrieve failed notifications", response = QueryResponse.class)
     public ResponseEntity<?> getFailedNotifications(
-            @RequestParam(value = "subscriptionName", required = true) final String subscriptionName,
+            @RequestParam(value = "subscriptionNames", required = true) final String subscriptionNames,
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
-            }
-            queryResponse.setQueryResponseEntity(queryResponseEntity);
-            LOGGER.debug("The response is : {}", response.toString());
-            if (processMissedNotification.deleteMissedNotification(subscriptionName)) {
-                LOGGER.debug("Failed notification for subscription {} was successfully removed from database",
-                        subscriptionName);
-             }
-            return new ResponseEntity<>(queryResponse, HttpStatus.OK);
-        } catch (Exception e) {
-            String errorMessage = "Failed to extract the data from the failed notification object based on subscription name "
-                    + subscriptionName + ".";
-            LOGGER.error(errorMessage, e);
+        List<String> subscriptionNameList = Arrays.asList(subscriptionNames.split(","));
+
+        JSONObject response = fetchFailedNotifications(subscriptionNameList);
+        JSONArray foundArray = response.getJSONArray(FOUND_NOTIFICATIONS);
+        JSONArray notFoundArray = response.getJSONArray(NOT_FOUND_NOTIFICATIONS);
+        HttpStatus httpStatus;
+        if (foundArray.length() > 0) {
+            httpStatus = HttpStatus.OK;
+        } else {
+            httpStatus = HttpStatus.NOT_FOUND;
+        }
+        if (httpStatus == HttpStatus.NOT_FOUND) {
+            String errorMessage = "Failed to fetch failed notifications for subscriptions:\n"
+                    + notFoundArray.toString();
             String errorJsonAsString = ResponseMessage.createJsonMessage(errorMessage);
-            return new ResponseEntity<>(errorJsonAsString, HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>(errorJsonAsString, httpStatus);
+        }
```

Comment:
I think you could even break out from the line 73 and forward to something like, "createResponse" and then do the logic to see which response code and error message it should contain. But this looks much better than before :)

Reply:
Done